### PR TITLE
Template quorum public key via k8s init containers, de-duplicate preprod and prod index files

### PR DESCRIPTION
**Pull request by r-n-o**

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -40,20 +40,23 @@ http {
...
-       }
+        }
     }
 
     server {
         listen 8081;
-        root /usr/share/nginx/export;
+        root /usr/share/nginx;
+        location / {
+            try_files export/$uri templated/export/index.hml;
```
Comment (andrewkmin) [inline nginx.conf:58]: tiny nit: `.html`

Comment (r-n-o) [reply nginx.conf]: Good catch 😮‍💨

File: import/index.template.html [hunk 2]
```diff
@@ -287,8 +287,12 @@
...
        */
       async function verifyEnclaveSignature(enclaveQuorumPublic, publicSignature, publicKey) {
-        /** Turnkey Signer enclave's public key (preprod) */
-        const TURNKEY_SIGNER_ENCLAVE_QUORUM_PUBLIC_KEY = "04f3422b8afbe425d6ece77b8d2469954715a2ff273ab7ac89f1ed70e0a9325eaa1698b4351fd1b23734e65c0b6a86b62dd49d70b37c94606aac402cbd84353212";
+        /** Turnkey Signer enclave's public keys */
+        const TURNKEY_SIGNERS_ENCLAVES = {
+          "prod": "04cf288fe433cc4e1aa0ce1632feac4ea26bf2f5a09dcfe5a42c398e06898710330f0572882f4dbdf0f5304b8fc8703acd69adca9a4bbf7f5d00d20a5e364b2569",
+          "preprod": "04f3422b8afbe425d6ece77b8d2469954715a2ff273ab7ac89f1ed70e0a9325eaa1698b4351fd1b23734e65c0b6a86b62dd49d70b37c94606aac402cbd84353212"
+        }
+        const TURNKEY_SIGNER_ENCLAVE_QUORUM_PUBLIC_KEY = TURNKEY_SIGNERS_ENCLAVES["${TURNKEY_SIGNER_ENVIRONMENT}"];
```
Comment (james-callahan) [inline import/index.template.html:295]: Should we throw an error here if not found?

File: kustomize/resources.yaml [hunk 3]
```diff
@@ -10,6 +10,29 @@ metadata:
...
+          env:
+            - name: TURNKEY_SIGNER_ENVIRONMENT
+              value: "prod"
+          command:
+            - sh
+            - -c
+            - |
+                mkdir -p templated/import && envsubst '${TURNKEY_SIGNER_ENVIRONMENT}' < import/index.template.html > templated/import/index.html;
+                mkdir -p templated/import && envsubst '${TURNKEY_SIGNER_ENVIRONMENT}' < import/standalone.template.html > templated/import/standalone.html;
+                mkdir -p templated/export && envsubst '${TURNKEY_SIGNER_ENVIRONMENT}' < export/index.template.html > templated/export/index.html;
```
Comment (james-callahan) [inline kustomize/resources.yaml:25]: Can do all the `mkdir` upfront
```suggestion
            - |
                mkdir -p templated/export templated/import;
                envsubst '${TURNKEY_SIGNER_ENVIRONMENT}' < import/index.template.html > templated/import/index.html;
                envsubst '${TURNKEY_SIGNER_ENVIRONMENT}' < import/standalone.template.html > templated/import/standalone.html;
                envsubst '${TURNKEY_SIGNER_ENVIRONMENT}' < export/index.template.html > templated/export/index.html;
```

File: nginx.conf [hunk 4]
```diff
@@ -40,20 +40,23 @@ http {
...
     # Prod
     server {
-       listen 8080;
-       root /usr/share/nginx/auth;
+        listen 8080;
+        root /usr/share/nginx/auth;
 
         # Health endpoint for k8s
-       location = /health {
+        location = /health {
```
Comment (james-callahan) [inline nginx.conf:47]: future improvement: lets move this endpoint elsewhere!

Comment (r-n-o) [reply nginx.conf:47]: Where do you think we should move it? Another port?

File: import/index.template.html (hunk 2 above)
Comment (r-n-o) [reply import/index.template.html:295]: Good call!

File: nginx.conf (hunk 4 above)
Comment (james-callahan) [reply nginx.conf:47]: > Another port?

Yeah probably. and/or find some sort of metrics plugin

**Review by james-callahan — APPROVED** [review]

**Pull request merged**